### PR TITLE
Lasb 585

### PR DESCRIPTION
**Pull request by hkhela**
## What

[Link to story] https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=454&projectKey=LASB&modal=detail&selectedIssue=LASB-585

Describe what you did and why.

Added functionality to create, update, delete, show hearing days in mock

## Checklist

Before you ask people to review this PR:

- [ ] Tests and linters should be passing
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by chris-groves — APPROVED** [review]

File: app/controllers/admin/hearing_days_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def hearing_day_attributes
+      %i[
+        id
+        sittingDay
+        listingSequence
+        listedDurationMinutes
+        startTime
+        endTime
+        court_centre_id
```
Comment (arthurashman) [inline app/controllers/admin/hearing_days_controller.rb:51]: court_room_id

File: app/controllers/admin/hearing_days_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      params.require(:hearing_day).permit(hearing_day_attributes)
+    end
+
+    def hearing_day_attributes
+      %i[
+        id
+        sittingDay
+        listingSequence
+        listedDurationMinutes
+        startTime
```
Comment (arthurashman) [inline app/controllers/admin/hearing_days_controller.rb:49]: These are no longer in the schema, apologies for previously showing you they were needed

File: app/controllers/admin/hearing_days_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        endTime
```
Comment (arthurashman) [inline app/controllers/admin/hearing_days_controller.rb:50]: These are no longer in the schema, apologies for previously showing you they were needed

File: app/controllers/admin/prosecution_cases_controller.rb [hunk 4]
```diff
@@ -108,7 +108,7 @@ def hearings_attributes
        :isEffectiveTrial,
        :isBoxHearing,
        { hearing_type_attributes: %i[id description] },
-       { hearing_days_attributes: %i[id sittingDay listedDurationMinutes] }]
+       { hearing_days_attributes: %i[id sittingDay listedDurationMinutes court_centre_id] }]
```
Comment (arthurashman) [inline app/controllers/admin/prosecution_cases_controller.rb:111]: court_room _id

File: app/views/admin/hearing_days/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    <%= f.label :listingSequence, 'Listing Sequence'%>
+    <%= f.text_field :listingSequence %>
+  </div>
+
+  <div class="field">
+    <%= f.label :listedDurationMinutes, 'Listed Duration Minutes' %>
+    <%= f.text_field :listedDurationMinutes %>
+  </div>
+
+  <div class="field">
```
Comment (arthurashman) [inline app/views/admin/hearing_days/_form.html.erb:23]: remove start time field

File: app/views/admin/hearing_days/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    <%= f.label :startTime, 'Start time' %>
+    <%= f.text_field :startTime %>
+  </div>
+
+  <div class="field">
+    <%= f.label :endTime, 'End time' %>
```
Comment (arthurashman) [inline app/views/admin/hearing_days/_form.html.erb:29]: remove end time field

File: app/views/admin/hearing_days/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    <%= f.text_field :endTime %>
+  </div>
+
+  <div class="field">
+    <%= f.label :court_centre_id, 'Court Centre' %>
+    <%= f.select :court_centre_id, HmctsCommonPlatform::Reference::CourtCentre.all.collect{ |cc| [cc.oucode_l3_name, cc.id ] } %>
+  </div>
+
```
Comment (arthurashman) [inline app/views/admin/hearing_days/_form.html.erb:37]: Add field for court_room_id - I don't think we have a list of these anywhere, so just make it a text field and factorybot can just put in a uuid

File: app/views/admin/prosecution_cases/_hearing.html.erb [hunk 8]
```diff
@@ -33,6 +33,7 @@
     <%= hearing_day.label :listedDurationMinutes %>
     <%= hearing_day.text_field :listedDurationMinutes %>
   </div>
+
```
Comment (arthurashman) [inline app/views/admin/prosecution_cases/_hearing.html.erb:36]: delete

File: spec/factories/hearing_days.rb [hunk 9]
```diff
@@ -6,4 +6,10 @@
     listingSequence { 1 }
     listedDurationMinutes { 1 }
   end
+
+  factory :realistic_hearing_day, class: "HearingDay" do
+    sittingDay { Faker::Date.forward(days: 23) }
+    listingSequence { Faker::Number.number(digits: 1) }
+    listedDurationMinutes { Faker::Number.number(digits: 3) }
```
Comment (arthurashman) [inline spec/factories/hearing_days.rb:13]: courtRoomId { SecureRandom.uuid }

File: spec/factories/hearing_days.rb [hunk 10]
```diff
@@ -6,4 +6,10 @@
     listingSequence { 1 }
     listedDurationMinutes { 1 }
```
Comment (arthurashman) [inline spec/factories/hearing_days.rb:7]: isCancelled { false }
courtCentreId { "003245ab-10cf-3145-8364-e30e72892d95" }
courtRoomId { "797ed295-9980-487e-af87-858e1f047d7d" }

File: spec/factories/hearing_days.rb (hunk 9 above)
Comment (arthurashman) [reply spec/factories/hearing_days.rb:13]: courtCentreId { HmctsCommonPlatform::Reference::CourtCentre.all.collect{ |cc|  cc.id }.sample }

File: app/views/admin/hearing_days/_form.html.erb (hunk 7 above)
Comment (arthurashman) [reply app/views/admin/hearing_days/_form.html.erb:37]: add checkbox for isCancelled
 
 <div class="field">
    <%= f.label :isCancelled, 'Is Cancelled' %>
    <%= f.check_box :isCancelled %>
  </div>

File: spec/factories/hearing_days.rb (hunk 9 above)
Comment (arthurashman) [reply spec/factories/hearing_days.rb:13]: isCancelled { Faker::Boolean.boolean }

**Review by chris-groves — APPROVED** [review]

**Review by arthurashman — APPROVED** [review]

**Pull request merged**